internal/builder/inspect.go

```diff
@@ -110,6 +118,11 @@ func (i *Inspector) Inspect(name string, daemon bool, orderDetectionDepth int) (
 		return Info{}, fmt.Errorf("calculating detection order: %w", err)
 	}
 
+	detectionOrderExtensions, err := i.detectionOrderCalculator.Order(orderExtensions, layers, orderDetectionDepth)
```

Comment:
This probably isn't necessary (as extensions cannot be nested)

Reply:
In the same vein as my comment above, should orderExtensions be passed directly?

Reply:
Per comment above, I think you could make a helper function like `toPubbldrDetectionOrder` that just converts to the new type.